### PR TITLE
[BenchGC] attach DLTI for mlir module

### PR DESCRIPTION
**Pull request by xurui1995**
* add code for getting the cpuinfo
*  provides `--cpu_cache_sizes` and `--max_vector_width` for user to add the dlti manually
* using env `GC_NUM_THREADS` to set the 'num_threads' of dlti

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -49,6 +49,7 @@ jobs:
       - name: Build and install benchgc
         working-directory: build
         run: |
+          pip install py-cpuinfo
```
Comment (WangJialei-A) [inline .github/workflows/build.yml:52]: please update the dependency in `setup.py`, not here

Comment (xurui1995) [reply .github/workflows/build.yml:52]: I already added in `setup.py`, but it got build error in ci

Comment (xurui1995) [reply .github/workflows/build.yml]: changed `py-cpuinfo`  to  `cpuinfo` in `setup.py`

Comment (xurui1995) [reply .github/workflows/build.yml]: fixed

File: test/benchgc/README.md [hunk 2]
```diff
@@ -2,32 +2,50 @@
...
+Benchgc is a tool used to verify the correctness and performance of graph compiler. Benchgc accepts MLIR files as test cases and prepares test data for them. For correctness verification, Benchgc will use PyTorch as a reference for comparison.
 
 ## Prerequisite
 * python >= 3.10
 * torch >= 2.2
-* pybind11
+* pip install py-cpuinfo
+* Enable mlir python binding, Refer to [`python/README.md`](../../python/README.md) for detail
+* Set the envs
+  * GC_NUM_THREADS [int] : the `num_threads` for dlti attr, default = 1
```
Comment (ZhennanQin) [inline test/benchgc/README.md:13]: Please use `OMP_NUM_THREADS` instead.

Comment (xurui1995) [reply test/benchgc/README.md]: fixed

**Comment by xurui1995** [comment]
Currently there is a issue for the lib cpuinfo on the CI machine, the  `l1_data_cache_size` is from the 'lscpu' output, and the value is `4.5 MiB` which can not be converted into int bytes. 
I am trying some new solutions for getting cache size.

File: test/benchgc/src/benchgc/__main__.py [hunk 3]
```diff
@@ -124,6 +124,20 @@ def add_common_options(parser: argparse.ArgumentParser):
         help="if we need print the ir during the pass-pipeline",
     )
 
+    parser.add_argument(
+        "--cpu_cache_sizes",
+        required=False,
+        help="set the cpu cache sizes, format: L1:L2:L3",
```
Comment (yifeizh2) [inline test/benchgc/src/benchgc/__main__.py:130]: Can you provide a detailed example for better understanding `set the cpu cache sizes in bytes, format: L1:L2:L3 (e.g. 32768:1048576:33554432)`.

File: test/benchgc/src/benchgc/mlir/util.py [hunk 4]
```diff
@@ -146,3 +148,49 @@ def get_kernel_func_from_module(
         if type(f) is func.FuncOp and str(f.name).strip('"') == func_name:
             return f
     raise ValueError("can not find the entry function")
+
+
+def attch_dlti(flags, module: ir.Module):
```
Comment (yifeizh2) [inline test/benchgc/src/benchgc/mlir/util.py:153]: Typo here?

File: test/benchgc/src/benchgc/__main__.py (hunk 3 above)
Comment (xurui1995) [reply test/benchgc/src/benchgc/__main__.py:130]: The Readme was updated for the new args.
### --cpu_cache_sizes, --max_vector_width
* BenchGC will automatically obtain target info and add the DLTI attr to the IR
* In some cases, if the system info obtained by BenchGC is not accurate, you can specify the relevant attributes for BenchGC through these options.
* --cpu_cache_sizes: cpu cache sizes in bytes, format: L1:L2:L3, example: `--cpu_cache_sizes 49152:2097152:110100480`
* --max_vector_width: the maximum width of vector registers available in a CPU, example `--max_vector_width `

**Review by yifeizh2 — APPROVED** [review]

**Review by zhczhong — APPROVED** [review]

**Pull request merged**